### ISSUE TITLE
Unable to Update Saga Document

### ISSUE THREAD

**Issue by kevin-whalley-gc** [sbmako/NServiceBus.MongoDB#16]
After upgrading to MongoCSharpDriver 2.0.1 and NServiceBus.MongoDb 5.2 we experienced total failure to update our sagas. The exception experienced was: "Unable to save with id ..."

https://github.com/sbmako/NServiceBus.MongoDB/blob/master/src/NServiceBus.MongoDB/SagaPersister/MongoSagaPersister.cs#L89

After pulling the source down and debugging we identified that the cast of saga.Id to a string was resulting in a failed query because our Id is a GUID and stored as a LUUID in MongoDb.

https://github.com/sbmako/NServiceBus.MongoDB/blob/master/src/NServiceBus.MongoDB/Extensions/DocumentVersionExtensions.cs#L59

The version of NServiceBus.MongoDb we updated from was 2.0.3 and the MongoCSharpDriver was 1.10.0. A hotfix would be much appreciated.

**Comment by sbmako** [comment]
Could you provide a sample of your saga data so I can add a unit test?

**Comment by kevin-whalley-gc** [comment]
```
{ 
    "_id" : BinData(3, "P3dSmnd110yn2qUAAOXjzA=="), 
    "_t" : "DeviceCommandSagaState", 
    "CorrelationId" : BinData(3, "DdqPKCAFvEOPjyXhiwFKMg=="), 
    "SerialId" : "luatestdevice002", 
    "FailureReason" : null, 
    "CommandName" : "GreatCall.Dms.Utilities.Contracts.Commands.PlayFindDeviceSound", 
    "FormattedMessage" : null, 
    "DeviceErrorCode" : null, 
    "DeviceErrorDescription" : null, 
    "Steps" : [
        {
            "Name" : NumberInt(0), 
            "Timestamp" : [
                NumberLong(635762194204061580), 
                NumberInt(0)
            ]
        }
    ], 
    "SagaKey" : "luatestdevice002_GreatCall.Dms.Utilities.Contracts.Commands.PlayFindDeviceSound", 
    "Originator" : "GreatCall.Dms.Utilities.AcceptanceTest.Endpoint", 
    "OriginalMessageId" : "bceac994-beba-44ef-8919-a50000e5e397", 
    "DocumentVersion" : NumberInt(0)
}

**Comment by sbmako** [comment]
Could you provide the saga data class?

**Comment by kevin-whalley-gc** [comment]
```c#
public sealed class DeviceCommandSagaState : SagaData
        {
        }
public abstract class SagaData : State, IContainSagaData, IHaveDocumentVersion
    {
        [Unique]
        public string SagaKey { get; set; }

        /// <summary>
        /// Gets or sets a value indicating the device update has failed
        /// </summary>
        public bool HasFailed
        {
            get { return this.FailureReason != null; }
        }

        //IContainSagaData properties
        public Guid Id { get; set; }
        public string Originator { get; set; }
        public string OriginalMessageId { get; set; }

        //IHaveDocumentVersion properties
        public int DocumentVersion { get; set; }
    }
```

**Comment by sbmako** [comment]
Fixed in v5.2.1 https://github.com/sbmako/NServiceBus.MongoDB/tree/5.2.1
